### PR TITLE
Better logging for mismatched applications

### PR DESCRIPTION
**Pull request by gabrielcocenza**
- Currently the logging does not include the unit, so it's harder to the user to know which unit didn't get upgrade.

### CODE REVIEW

File: cou/apps/core.py [hunk 1]
```diff
@@ -239,15 +240,22 @@ def current_os_release(self) -> OpenStackRelease:
...
+            return next(iter(os_versions))
 
-        if len(os_versions) == 1:
-            return os_versions.pop()
         # NOTE (gabrielcocenza) on applications that use single-unit or paused-single-unit
         # upgrade methods, more than one version can be found.
+        mismatched_repr = [
+            f"'{openstack_release.codename}': {os_versions[openstack_release]}"
+            for openstack_release in os_versions
+        ]
```
Comment (rgildein) [inline cou/apps/core.py:255]: not-blocking:
```suggestion
        mismatched_repr = [
            f"'{openstack_release.codename}': {apps}"
            for openstack_release, apps in os_versions.items()
        ]
```

**Review by rgildein — APPROVED** [review]
One small comment, other than that LGTM

**Review by rgildein — APPROVED** [review]
LGTM

File: cou/apps/core.py [hunk 2]
```diff
@@ -239,15 +240,22 @@ def current_os_release(self) -> OpenStackRelease:
...
+        if len(os_versions.keys()) == 1:
+            return next(iter(os_versions))
 
-        if len(os_versions) == 1:
-            return os_versions.pop()
         # NOTE (gabrielcocenza) on applications that use single-unit or paused-single-unit
         # upgrade methods, more than one version can be found.
+        mismatched_repr = [
+            f"'{openstack_release.codename}': {apps}"
+            for openstack_release, apps in os_versions.items()
```
Comment (aieri) [inline cou/apps/core.py:254]: s/apps/units/ right?

**Review by aieri — COMMENTED** [review]
overall lgtm, just one question

File: cou/apps/core.py (hunk 2 above)
Comment (rgildein) [reply cou/apps/core.py]: My suggestion was not correct, sorry.

**Review by rgildein — APPROVED** [review]

**Review by aieri — APPROVED** [review]

**Pull request merged**